diff --git a/plugins/clish/hook_log.c b/plugins/clish/hook_log.c
index cafd29e..6ec171e 100644
--- a/plugins/clish/hook_log.c
+++ b/plugins/clish/hook_log.c
@@ -11,6 +11,7 @@
 #include <pwd.h>
 
 #include "clish/shell.h"
+#include "clish/plugin/mgmt_clish_utils.h"
 
 #define SYSLOG_IDENT "klish"
 
@@ -37,8 +38,18 @@ CLISH_HOOK_LOG(clish_hook_log)
 		if (!(uname = getenv("LOGNAME")))
 			uname = user ? user->pw_name : "unknown";
 	}
-	syslog(LOG_INFO, "%u(%s) %s : %d",
-		user ? user->pw_uid : getuid(), uname, line, retcode);
+
+        int facility = LOG_LOCAL4;
+        char *masked_line = NULL;
+
+        mask_password(line, &masked_line);
+
+        if (masked_line) {
+            syslog(LOG_INFO|facility, "User \"%s\" command \"%s\" status - %s",
+                   uname, masked_line, retcode ? "failure" : "success");
+            free(masked_line);
+            masked_line = NULL;
+        }
 
 	return 0;
 }
